src/main/java/net/minecraftforge/client/model/animation/ModelBlockAnimation.java

```diff
@@ -390,11 +391,25 @@ public TRSRTransformation apply(float time)
                         case ZS:
                             scale.z = value;
                             break;
+                        case XORIGIN:
+                            origin.x = value;
+                            break;
+                        case YORIGIN:
+                            origin.y = value;
+                            break;
+                        case ZORIGIN:
+                            origin.z = value;
+                            break;
                     }
                 }
                 Quat4f rot = new Quat4f();
                 rot.set(rotation);
-                return TRSRTransformation.blockCenterToCorner(new TRSRTransformation(translation, rot, scale, null));
+                TRSRTransformation base = new TRSRTransformation(translation, rot, scale, null);
+                Vector3f negOrigin = (Vector3f) origin.clone();
```

Comment:
Vector3f has a copy method or copy constructor I'm pretty sure which is better than cloning